### DISCUSSION TITLE
Formatting in `.md`-files?

### DISCUSSION THREAD

**Discussion by felix-bohlin** [vuejs/vitepress#4388] [Q&A] [answered]
I'm really struggling with getting formatting in `.md` files to work properly. The HTML nesting indentation is way off and when I try to fix it manually the code just becomes invalid and Vite won't parse it.

Should it work straight out of the box with a specific VSCode extension and VSCode setting turned on? I'm hoping I'm missing something that's been staring me in the face.

Currently I use Prettier to format my `.md`-files because that keeps Vite from breaking less than if I used the Vue extension for that.

Thanks in advance!

**Comment by brc-dd** [comment, marked as answer]
Use prettier + markdownlint extensions. Don't use volar's formatting. They should work for most things, but they still don't format html/vue code inside markdown files (https://github.com/prettier/prettier/issues/8480). I had written a [patch](https://gist.github.com/brc-dd/5df39b2db587bd7f7b9a7d8016934736) to support vue in markdown but that only worked with prettier 2. I'm not much familiar with writing prettier plugins myself, but someone having experience in that field might create a prettier plugin for html/vue in markdown.

**Reply by felix-bohlin** [reply]
Thanks for your reply! I'll mark this as answered since there's no clean way to solve HTML in markdown yet.

**Reply by brc-dd** [reply]
@felix-bohlin Converted that patch to plugin, not sure how correct it is, but seems to work fine for me. - https://github.com/brc-dd/prettier-plugin-vitepress

**Reply by felix-bohlin** [reply]
Cool idea! It still doesn't really work for me on my first try, but I'll come back with a proper analysis in a bit.

It re-formatted things in a way that made the markdown parser confused and thought it was missing end tags that were there.

**Reply by brc-dd** [reply]
Ah, ok 😅 If possible can you share the things it formats weirdly?

Prettier internally uses remark instead of markdown-it, so there will be some difference because of it 👀

**Reply by felix-bohlin** [reply]
Ok, so I'm not really sure what happens 😅 But this plugin is __very__, VERY strict! 😅 If there's an extra line break somewhere the formatter will think the elements above and under aren't closed, so it goes around and creates and removes elements in order to fix that (I think). I had to manually redo a bunch of stuff because it got pretty messy but will return to this and see what actually went wrong.

Inside an ```html -block it works great though! 😄

**Reply by brc-dd** [reply]
Ok I see the issue on your ui repo. It's behaving weird when there is `<template #slot>`. I'll fix that and let you know.

**Reply by felix-bohlin** [reply]
>But there shouldn't be new line between html tags

Is that a known restriction when writing HTML in markdown files or a Vue slot rule I'm unaware of?

**Reply by brc-dd** [reply]
Ah, never mind. Ignore that. It should've work even with the newlines. I'll fix the plugin.
